Intersect.Client/Entities/Player.cs

```diff
@@ -2388,27 +2388,31 @@ protected override bool ShouldDrawHpBar
                     return true;
                 }
 
-                if (Id == Globals.Me.Id)
+                var me = Globals.Me;
+                bool userPreference = false;
+
+                if (Globals.Database.MyOverheadHpBar && Id == me.Id)
                 {
-                    return Globals.Database.MyOverheadHpBar;
+                    userPreference = true;
                 }
-
-                if (Globals.Database.PartyMemberOverheadHpBar && Globals.Me.IsInMyParty(this))
+                else if (Globals.Database.PlayerOverheadHpBar && Id != me.Id)
                 {
-                    return true;
+                    userPreference = true;
                 }
-
-                if (Globals.Database.FriendOverheadHpBar && Globals.Me.IsFriend(this))
+                else if (Globals.Database.PartyMemberOverheadHpBar && me.IsInMyParty(this))
                 {
-                    return true;
+                    userPreference = true;
                 }
-
-                if (Globals.Database.GuildMemberOverheadHpBar && Globals.Me.IsGuildMate(this))
+                else if (Globals.Database.FriendOverheadHpBar && me.IsFriend(this))
                 {
-                    return true;
+                    userPreference = true;
+                }
+                else if (Globals.Database.GuildMemberOverheadHpBar && me.IsGuildMate(this))
+                {
+                    userPreference = true;
                 }
 
-                return Globals.Database.PlayerOverheadHpBar;
+                return userPreference;
```

Comment:
Get rid of this extra variable, and all of the `else` keywords.

Just return true (and return <the last if statement>)